src/UserSpaceInstrumentation/TrampolineTest.cpp

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_split.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include "AllocateInTracee.h"
+#include "Attach.h"
+#include "ElfUtils/LinuxMap.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "Trampoline.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace internal {
+
+namespace {
+
+using absl::numbers_internal::safe_strtou64_base;
+using orbit_base::ReadFileToString;
+
+extern "C" __attribute__((noinline)) int DoubleAndIncrease(int i) {
+  i = 2 * i;
+  return i + 1;
+}
+
+}  // namespace
+
+TEST(TrampolineTest, DoAddressRangesOverlap) {
```

Comment:
Add tests with extremes touching, e.g., {1,3},{3,7} and {3,7},{7,9}

Reply:
done